internal/permissions/groups/middleware.go

```diff
@@ -0,0 +1,81 @@
+package userpermissions
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/go-chi/chi"
+	"github.com/sirupsen/logrus"
+	"github.com/source-academy/stories-backend/internal/auth"
+	"github.com/source-academy/stories-backend/internal/database"
+	apierrors "github.com/source-academy/stories-backend/internal/errors"
+	"github.com/source-academy/stories-backend/model"
+)
+
+type contextKey string
+
+const (
+	userGroupKey contextKey = "user_group_context"
+)
+
+func InjectUserGroupIntoContext(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		groupIDStr := chi.URLParam(r, "groupID")
+		groupID, err := strconv.Atoi(groupIDStr)
+		if err != nil {
+			apierrors.ServeHTTP(w, r, apierrors.ClientBadRequestError{
+				Message: fmt.Sprintf("Invalid groupID: %v", err),
+			})
```

Comment:
Should be 422 Unprocessable Entity instead of 400 Bad Request. Bad Request means incorrect request format (cannot be parsed). This is parsed but invalid.